europy/lifecycle/markdowner.py

```diff
@@ -0,0 +1,144 @@
+import os
+import errno
+from typing import Any, List
+from europy import report_directory
+
+
+class Markdown:
+
+    cls_dict_markdown = ""
+    tab = "  "
+    list_tag = '* '
+    htag = '#'
+
+    def __init__(self):
+        self.content = ""
+
+    def saveToFile(self, file_name: str):
+        file_path = os.path.join(report_directory, file_name)
+        if not os.path.exists(file_path):
+            try:
+                os.makedirs(os.path.dirname(file_path), exist_ok=True)
+            except OSError as exc:
+                if exc.errno != errno.EEXIST:
+                    raise
+        file = open(file_path, "w")
+        file.write(self.content)
+        file.close()
+
+    def add_linebreak(self):
+        self.content += self.create_block("", 1)
+        return self
+
+    def add_list_item(self, text: str, depth: int = 0):
+        intent = ""
+        if depth > 0:
+            intent = "".join([" " * 2] * depth)
+
+        self.content += self.create_block(intent + "- {}".format(text))
+        return self
+
+    def add_header(self, text: str, htype: int = 1):
+        string = "".join(["#"] * htype) + " {t}".format(t=text)
+        self.content += self.create_block(string, 2)
+        return self
+
+    def add_text(self, text: str):
+        self.content += self.create_block(text, 2)
+        return self
+
+    def add_horizontal_line(self):
+        self.content += self.create_block("___")
+        return self
+
+    def add_block_content(self, *lines: Any):
+        _lines: List[str] = list(lines)
+        self.content += self.create_block("> " + "  \n".join(_lines), 2)
+        return self
+
+    def add_image(self, file_name: str, alt_text: str):
+        file_path = os.path.join(report_directory+'/figures/', file_name)
```

Comment:
You should not include the report directory here. The path to the image file should be relative to the markdown file so adding in `report_directory + '/figures/'` is actually breaking the image rendering for me when I view the markdown locally.

Reply:
yes, you are right. this path has to be relative to the report file getting written to, that is why I picked up the report_directory which will be passed to the markdown function. at the time of writing the code, I noticed the image location was still being worked on, so added a temp snippet. the final function would take in the report_directory and append +/figures/+ like shown above and read the image from the relative path to display the image, let me know if this helps

Reply:
on further thought, I think I am going to make the function generic with the file path and file name as input param so the calling function can just give the name and path